test/ASTGen/keypath.swift

```diff
@@ -0,0 +1,243 @@
+// REQUIRES: swift_swift_parser
+// REQUIRES: swift_feature_ParserASTGen
+
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -dump-ast -enable-experimental-feature ParserASTGen -verify > %t/astgen.ast.raw
+// RUN: not %target-swift-frontend %s -dump-ast > %t/cpp-parser.ast.raw
```

Comment:
Comparing `-dump-ast` instead of `-dump-parse` because we generate `KeyPathExpr` differently.
In `libParse` the root expression is parsed as a normal expression, but in `SwiftParser` it's parsed as a `TypeSyntax`, so we generate it as `TypeExpr`.

Reply:
Would it be worth adding this as a comment to the test?